Consul.Test/SemaphoreTest.cs

```diff
@@ -363,5 +365,61 @@ public async Task Semaphore_Conflict()
 
             await semaphoreLock.Destroy();
         }
+
+        [Fact]
+        public async Task Passing_Cancelled_CancellationToken_Should_Throw_LockNotHeldException()
+        {
+            const string keyName = "service/myApp/leader";
+
+            var distributedLock = _client.CreateLock(new LockOptions(keyName)
+            {
+                SessionTTL = TimeSpan.FromSeconds(DefaultSessionTTLSeconds),
+                LockWaitTime = TimeSpan.FromSeconds(LockWaitTimeSeconds)
+            });
+
+            var cts = new CancellationTokenSource();
+
+            cts.Cancel();
+
+            await Assert.ThrowsAsync<LockNotHeldException>(async () => await distributedLock.Acquire(cts.Token));
+        }
+
+        [Fact]
+        public async Task Cancelling_A_Token_When_Acquiring_A_Lock_Should_Throw_LockNotHeldException()
+        {
+            const string keyName = "service/myApp/leader";
+
+            var masterClient = new ConsulClient(c =>
+            {
+                c.Token = TestHelper.MasterToken;
+                c.Address = TestHelper.HttpUri;
+            });
+
+            var distributedLock2 = masterClient.CreateLock(new LockOptions(keyName)
+            {
+                SessionTTL = TimeSpan.FromSeconds(DefaultSessionTTLSeconds),
+                LockWaitTime = TimeSpan.FromSeconds(LockWaitTimeSeconds)
+            });
+
+            var distributedLock = _client.CreateLock(new LockOptions(keyName)
+            {
+                SessionTTL = TimeSpan.FromSeconds(DefaultSessionTTLSeconds),
+                LockWaitTime = TimeSpan.FromSeconds(LockWaitTimeSeconds)
+            });
+
+            var cts = new CancellationTokenSource(TimeSpan.FromSeconds(1));
+
+            await distributedLock2.Acquire(); // Become "Master" with another instance first
+            var task = distributedLock.Acquire(cts.Token);
+
+            await Assert.ThrowsAsync<LockNotHeldException>(async () => await distributedLock.Acquire(cts.Token));
```

Comment:
`distributedLock.Acquire(cts.Token)` is now called twice - I don't think you need the ```var task = distributedLock.Acquire(cts.Token);```

Reply:
done